### PR TITLE
Added two new TypeAhead versions to the styleguide

### PR DESCRIPTION
**Pull request by jupiter007**
## Description

We will need TypeAhead/Autosuggest components for pages like the "Register" page where users have to search for an Institution.

- I added two TypeAhead components to the app: the first one does not include an "Other" option and the second one has it, and expects a "setOtherField" method to be passed to it that can be set to true when a user selects the "Other" field. That way the calling component can know to show the additional "Other" field.

- I remove the files related to the pilot project dmp landing page because it was breaking the build, and was added primarily for testing purposes.
- I added the use of graphql-codegen to generate the types for the graphql queries.
- Updated README.md file

Fixes # [10](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/10)

## Type of change
Please delete options that are not relevant
- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
The new TypeAheads were tested against the backend server GraphQL API for `affiliations` after they were added to the styleguide.

- Start the app along with the dmsp_backend_prototype `development` branch.
- Navigate to http://localhost:3000/styleguide and scroll down to the `Forms fields` section.
- Some tests to run
-- Enter "alaska" into input field. Should show results in dropdown
-- Select an option from the dropdown either using "Click" or "Enter" key. The input value to change to the selected one
-- Navigate through the dropdown using the up and down arrow keys
-- Enter a nonsensical value like "asdfadf". Try and tab out of the field. You should be restricted from doing so.

Also, unit tests for the components were added.

## Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I updated the CHANGELOG.md and added documentation if necessary
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have completed manual or automated accessibility testing for my changes
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/client-graphql-test/page.tsx [hunk 1]
```diff
@@ -13,6 +11,11 @@ query ContributorRoles{
     }
 }`;
 
+type ContributorRoles = {
```
Comment (briri) [inline app/client-graphql-test/page.tsx:14]: Could you setup codegen instead that auto-generates the types for you based on the GraphQL schema definition in the backend?
example `codegen.ts`
```
import type { CodegenConfig } from "@graphql-codegen/cli";

const config: CodegenConfig = {
  schema: "../dmsp_backend_prototype/src/schemas/*.ts",
  generates: {
    "./src/types.ts": {
      plugins: ["typescript", "typescript-resolvers"],
      config: {
        contextType: "../dmsp_backend_prototype/src/context#MyContext",
        enumsAsTypes: true,
      },
    },
  },
};

export default config;
```

File: components/Spinner/index.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    className: string;
+}
+function Spinner(props: SpinnerProps) {
+    return (
+        <>
+            {props?.isActive && (
+                <div
+                    id={(props.id ? props.id : "spinner")}
+                    tabIndex={-1}
+                    aria-hidden="true"
```
Comment (briri) [inline components/Spinner/index.tsx:16]: this is just a question, but how does a screen reader let the user know that the site is loading/processing?

**Review by briri — APPROVED** [review]
This looks good to me @jupiter007 

I just had one suggestion about using codegen to import the types from the backend codebase.

A we going to allow for both Typeaheads or are we waiting for consensus on which one is preferred?

File: app/client-graphql-test/page.tsx (hunk 1 above)
Comment (jupiter007) [reply app/client-graphql-test/page.tsx:14]: @briri Yes that's a good idea. I'll create a separate ticket for that.

File: components/Spinner/index.tsx (hunk 2 above)
Comment (jupiter007) [reply components/Spinner/index.tsx:16]: Good point. I should probably add some hidden text that is readable by the screen reader that says "loading".

File: app/client-graphql-test/page.tsx (hunk 1 above)
Comment (jupiter007) [reply app/client-graphql-test/page.tsx:14]: @briri I created this ticket to add the use of graphql-codegen: https://github.com/CDLUC3/dmsp_frontend_prototype/issues/88

Also, I added a "Loading..." message for screen readers in the TypeAheads.

**Comment by jupiter007** [comment]
@briri I merged the graphql-codegen changes to this PR. Could you take a quick look at the last two commits when you get the chance? Thanks.

**Pull request merged**